org.eclipse.scanning.malcolm.core/src/org/eclipse/scanning/malcolm/core/MalcolmEventDelegate.java

```diff
@@ -66,21 +45,22 @@ public void sendEvent(MalcolmEventBean event)  throws Exception {
 
 
 	public void addMalcolmListener(IMalcolmListener<MalcolmEventBean> l) {
-		if (mlisteners==null) mlisteners = Collections.synchronizedCollection(new LinkedHashSet<IMalcolmListener<MalcolmEventBean>>());
-		mlisteners.add(l);
+		if (listeners==null) listeners = Collections.synchronizedCollection(new LinkedHashSet<IMalcolmListener<MalcolmEventBean>>());
+		listeners.add(l);
 	}
 
 	public void removeMalcolmListener(IMalcolmListener<MalcolmEventBean> l) {
-		if (mlisteners==null) return;
-		mlisteners.remove(l);
+		if (listeners==null) return;
+		listeners.remove(l);
 	}
 
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00112)